### DISCUSSION TITLE
API - Endpoints updates

### DISCUSSION THREAD

**Discussion by mislavmatijevic** [foivz/pop-app-backend#3] [General]
Initial version of API endpoints is released in the [SRS](https://github.com/foivz/pop-app-backend/wiki/SRS#212-endpoints) part of the Wiki. It's got more than 30 different routes! 👀 Check it out! 🚀

**Comment by mislavmatijevic** [comment]
Endpoints have been updated. Dear @antoniokunstek, please take a careful look and get back to me on whether that's ok now or not. Our conversation yesterday made me notice a lot of my mistakes, so hopefully now it's cleaner. I kept in mind what you emphasized, on how each functionality needs a clean and compact controller. 👍🏼

**Reply by antoniokunstek** [reply]
Dear Mislav (@mislavmatijevic),

You've done an excellent job yet again. I reviewed all the routes and only two routes remained questionable. I am sending a picture as an attachment to this comment so that you can check it more easily.
I am not sure how much it is necessary to send the userId as a path parameter in the setRole and assignRole routes. If it is an already registered user, we can put this information inside the jwt token and it will be sent with the header, unless you want the possibility that the admin can also change the role or add a store to another user at any time. This is a question of business logic and I hope my question makes sense.

Do not hesitate to ask any question or comment, if I have not explained something well.

I thank you in advance for your answer and wish you a pleasant rest of the day.
Best regards!
Antonio Kunštek

P.S. I hope that you will use the upcoming vacation for rest, and not for programming or working on a project, we have the whole year for that.

![image](https://github.com/foivz/pop-app-backend/assets/61520141/1b2b0b7e-92fd-48b2-a1d5-d4dc18a933f6)

**Reply by mislavmatijevic** [reply]
Hello @antoniokunstek, thanks a lot on a detailed response!

***

Firstly, you correctly assumed my intentions with the Admin role. My idea is that the Admin can (one day) also change some user's role/store if needed. Also, it looks a lot more RESTfulish than a POST or PATCH request to `/api/v2/role` (since "role" is not really a resource of any kind - just a piece of user's defining attributes). Same goes for assigning stores - there, such a route would even cause confusion since POST to stores does already exist for creating stores as resources and PATCHing stores would cause confusion with updating store's information.

Of course, it would be possible to create new routes completely, e.g. `/api/v2/assign/role` and `/api/v2/assign/store`. I don't know how RESTful that is exactly, so I ask you for your opinion. That could be a road we take.

***

Secondly, The idea of putting user's role in the JWT sounds really good. However, that data would be incorrect unless we'd force a user to get a new JWT after changing role.

***

Please comment if you agree or disagree - it's possible I overlooked something in a hurry.

Yours truly
Mislav
